### PR TITLE
Remove kube-rbac-proxy from EKS-A codebase

### PR DESCRIPTION
**Pull request by abhay-krishna**
CAPI removed `kube-rbac-proxy` from its `Deployment`s during the `v1alpha3` to `v1alpha4` migration, and the various infrastructure providers also followed suit. This PR removes kube-rbac-proxy from the release bundle and also references in the code for `clusterctl` overrides.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: release/api/v1alpha1/bundle_types.go [hunk 1]
```diff
@@ -147,31 +147,27 @@ type CertManagerBundle struct {
 type CoreClusterAPI struct {
 	Version    string   `json:"version"`
 	Controller Image    `json:"controller"`
-	KubeProxy  Image    `json:"kubeProxy"`
```
Comment (g-gaston) [inline release/api/v1alpha1/bundle_types.go:150]: This is a breaking change. I would just leave all the fields as they are, add a `ommitempty` tag and mark as deprecated with a doc comment

Comment (abhay-krishna) [reply release/api/v1alpha1/bundle_types.go:150]: Makes sense. Can we just retain the API struct for backward compatibility but remove its references from everywhere else in the codebase?

**Pull request closed without merging**